Content.Server/DeviceLinking/Systems/PressurePlateSystem.cs

```diff
@@ -0,0 +1,114 @@
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<PressurePlateComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<PressurePlateComponent, StepTriggerAttemptEvent>(HandleTriggerAttempt);
+            SubscribeLocalEvent<PressurePlateComponent, EndCollideEvent>(OnEndCollide);
+            SubscribeLocalEvent<PressurePlateComponent, StartCollideEvent>(OnStartCollide);
+        }
+
+        private void OnInit(EntityUid uid, PressurePlateComponent component, ComponentInit args)
+        {
+            _signalSystem.EnsureSourcePorts(uid, component.PressedSignal, component.ReleasedSignal);
+        }
+
+        private void HandleTriggerAttempt(EntityUid uid, PressurePlateComponent component, ref StepTriggerAttemptEvent args)
+        {
+            args.Continue = true;
+        }
+        private void OnStartCollide(EntityUid uid, PressurePlateComponent component, ref StartCollideEvent args)
+        {
+
+            var otherUid = args.OtherEntity;
+            if (!args.OtherFixture.Hard)
+                return;
+            if (component.Colliding.Add(otherUid))
+            {
+                UpdateState(uid, component);
+                Dirty(uid, component);
+            }
+
+        }
+
+        private void OnEndCollide(EntityUid uid, PressurePlateComponent component, ref EndCollideEvent args)
+        {
+            var otherUid = args.OtherEntity;
+
+            if (!component.Colliding.Remove(otherUid))
+            {
+                return;
+            }
+            UpdateState(uid, component);
+            Dirty(uid, component);
+        }
+
+        private void UpdateState(EntityUid uid, PressurePlateComponent component)
+        {
+            var totalMass = 0f;
+            foreach (var ent in component.Colliding)
+            {
+                if (Deleted(ent))
+                {
+                    component.Colliding.Remove(ent);
+                    continue;
+                }
+                totalMass += GetEntWeightRecursive(ent);
+            }
+            if (component.State == PressurePlateState.Pressed && totalMass < component.WeightRequired) //Release
+            {
+                component.State = PressurePlateState.Released;
+                _signalSystem.InvokePort(uid, component.ReleasedSignal);
+
+                _audio.PlayPvs(component.ReleasedSound, uid);
+            }
+            if (component.State == PressurePlateState.Released && totalMass > component.WeightRequired) //Press
+            {
+                component.State = PressurePlateState.Pressed;
+                _signalSystem.InvokePort(uid, component.PressedSignal);
+                _audio.PlayPvs(component.PressedSound, uid);
+            }
+
+            if (TryComp(uid, out AppearanceComponent? appearance))
```

Comment:
redundant since SetData will `Resolve(uid, ref appearance, false)` just leave out appearance from the call

Reply:
just delete it?

Reply:
yes delete the if ()
unindent SetData
and remove `, appearance` argument